src/main/kotlin/de/bund/digitalservice/useid/config/WidgetSecurityHeadersFilter.kt

```diff
@@ -2,36 +2,37 @@ package de.bund.digitalservice.useid.config
 
 import de.bund.digitalservice.useid.tenant.REQUEST_ATTR_TENANT
 import de.bund.digitalservice.useid.tenant.Tenant
+import de.bund.digitalservice.useid.widget.FALLBACK_PAGE
 import de.bund.digitalservice.useid.widget.WIDGET_PAGE
 import jakarta.servlet.FilterChain
 import jakarta.servlet.http.HttpServletRequest
 import jakarta.servlet.http.HttpServletResponse
 import org.springframework.http.HttpHeaders
 import org.springframework.http.server.PathContainer
 import org.springframework.web.filter.OncePerRequestFilter
-import org.springframework.web.util.pattern.PathPattern
 import org.springframework.web.util.pattern.PathPatternParser
 import java.util.UUID
 
 internal const val HTTP_HEADER_CONTENT_SECURITY_POLICY = "Content-Security-Policy"
 class WidgetSecurityHeadersFilter : OncePerRequestFilter() {
 
-    private val widgetPagePath: PathPattern = PathPatternParser().parse("/$WIDGET_PAGE")
-    private val listOfPages = listOf(widgetPagePath)
-
     override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
-        val pathIsValidWidgetPages = listOfPages.any {
-            it.matches(PathContainer.parsePath(request.servletPath))
+        if (requestPathMatches(WIDGET_PAGE, request)) {
+            val tenant = request.getAttribute(REQUEST_ATTR_TENANT) as Tenant
+            tenant.cspNonce = UUID.randomUUID().toString()
+            response.setHeader(HTTP_HEADER_CONTENT_SECURITY_POLICY, WidgetContentSecurityPolicy.headerValue(tenant.cspHost, tenant.cspNonce))
+            response.setHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, tenant.cspHost)
+            response.setHeader(HttpHeaders.VARY, HttpHeaders.ORIGIN)
         }
 
-        if (!pathIsValidWidgetPages) return filterChain.doFilter(request, response)
-
-        val tenant = request.getAttribute(REQUEST_ATTR_TENANT) as Tenant
-        tenant.cspNonce = UUID.randomUUID().toString()
-        response.setHeader(HTTP_HEADER_CONTENT_SECURITY_POLICY, WidgetContentSecurityPolicy.headerValue(tenant.cspHost, tenant.cspNonce))
-        response.setHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, tenant.cspHost)
-        response.setHeader(HttpHeaders.VARY, HttpHeaders.ORIGIN)
+        if (requestPathMatches(FALLBACK_PAGE, request)) {
+            response.setHeader(HTTP_HEADER_CONTENT_SECURITY_POLICY, WidgetContentSecurityPolicy.headerValueNoFrame())
```

Comment:
We probably need the nonce logic here as well, since we execute inline js. I posted the error in slack in the thread.   
If you run the app locally with the e-service-example, and you go onto the fallback page, you will see the error in your console.

Reply:
<img width="725" alt="image" src="https://user-images.githubusercontent.com/29574225/230317314-7507575c-e1c5-4c4c-9f38-1f64467d179f.png">

Reply:
You're right. Thought it had something to do with the widget-script, but it's our inline script, right.